app/src/main/java/com/zulip/android/activities/MessageListFragment.java

```diff
@@ -290,6 +305,81 @@ private void selectPointer() {
         }
     }
 
+    /**
+     * Edit a message passed as parameter
+     *
+     * @param message Message to be edited
+     */
+    private void editMessage(final Message message, final int position) {
+        int maxMessageContentEditLimit = app.
+                getSettings().
+                getInt(Constants.MAXIMUM_CONTENT_EDIT_LIMIT, 1800);
+        int timeSinceMessageSend = (int) ((System.currentTimeMillis() - message.getTimestamp().getTime()) / 1000);
+
+        if (timeSinceMessageSend > maxMessageContentEditLimit) {
+            Toast.makeText(getContext(), R.string.maximum_time_limit_error, Toast.LENGTH_SHORT).show();
+        } else {
+            final View dialogView = View.inflate(getContext(),
+                    R.layout.message_edit_dialog, null);
+            //Pop up a dialog box with previous message content
+            final AlertDialog dialog = new AlertDialog.Builder(getContext())
+                    .setTitle(R.string.edit_message)
+                    .setView(dialogView)
+                    .setPositiveButton(android.R.string.ok, null)
+                    .setNegativeButton(android.R.string.cancel, null)
+                    .create();
+            dialog.getWindow().setSoftInputMode(
+                    WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE);
+            dialog.show();
+
+            final EditText question = (EditText) dialogView.findViewById(R.id.quiz_question);
+            question.setText(message.getContent().trim());
+
+            //Move cursor to end of text
+            question.setSelection(question.getText().length());
+
+            //OK button listener
+            dialog.getButton(DialogInterface.BUTTON_POSITIVE).setOnClickListener(new View.OnClickListener() {
+                //Show edit option only on if current user send it.
+                @Override
+                public void onClick(View view) {
+                    dialog.cancel();
+                    InputMethodManager imm = (InputMethodManager) getActivity().getSystemService(Context.INPUT_METHOD_SERVICE);
+                    imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
+                    //Start a progress dialog indicating editing message
+                    final ProgressDialog progress = new ProgressDialog(getActivity());
+                    progress.setCancelable(false);
+                    progress.setMessage(ZulipApp.get().getString(R.string.editing_message));
```

Comment:
ZulipApp class is used to store global configuration for the app. An instance of this class is generally present in all classes. Here i see `app` data member present in the class and could be used instead of ZulipApp.get().